### PR TITLE
Respect users name display settings

### LINKED ISSUE #1158: Order for Japanese names
Hi,

I am using Signal in Japanese and noticed that the family name and the given name are displayed in the wrong order. For Japanese names written in Japanese, the family name should come before the given name. Moreover, the space between the family name and the given name can be omitted.

For example, if the given name is G and the family name is F, Signal currently shows
G F
but it should be either
F G
or
FG

FG (the space being omitted) is better if it is written in a sentence (e.g., "Missed a call from xxx") since Japanese sentences don't use space. It is safer to show FG everywhere, which is what is done in  Apple's Mail and iMessage.

Since the reverse order appears quite awkward and may give people a bad impression, I would like to ask you to look into this problem.

The problem appears in iPhone 6s with iOS 9.3.1 and Signal 2.2.

Thank you.

### PR DESCRIPTION
**Pull request by abcdev**
### First time contributor checklist

- [x] I have read the [README](https://github.com/WhisperSystems/Signal-iOS/blob/master/README.md) and [CONTRIBUTING](https://github.com/WhisperSystems/Signal-iOS/blob/master/CONTRIBUTING.md) documents
- [x] I have signed the [Contributor Licence Agreement](https://whispersystems.org/cla/)
### Contributor checklist

- [x] I'm following the [code, UI and style conventions](https://github.com/WhisperSystems/Signal-iOS/blob/master/CONTRIBUTING.md#code-conventions)
- [x] My commits are rebased on the latest master branch
- [x] My commits are in nice logical chunks
- [x] My contribution is fully baked and is ready to be merged as is
- [x] I have tested my contribution on these devices:
  - iPhone 5s, iOS 10 beta 4
- [x] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards)

---
### Description

Fixes #1158 by respecting users name display settings. Depends on [WhisperSystems/SignalServiceKit/pull/22](https://github.com/WhisperSystems/SignalServiceKit/pull/22)

### CODE REVIEW

File: Signal/src/views/ContactTableViewCell.m [hunk 1]
```diff
@@ -30,34 +30,53 @@ - (NSAttributedString *)attributedStringForContact:(Contact *)contact {
     NSMutableAttributedString *fullNameAttributedString =
         [[NSMutableAttributedString alloc] initWithString:contact.fullName];
 
+    BOOL firstNameDisplay = ABPersonGetCompositeNameFormat() == kABPersonCompositeNameFormatFirstNameFirst ? YES : NO;
+    BOOL firstNameSortOrdering = ABPersonGetSortOrdering() == kABPersonCompositeNameFormatFirstNameFirst ? YES : NO;
```
Comment (michaelkirk) [inline Signal/src/views/ContactTableViewCell.m]: `ABPersonGetSortOrdering` doesn't return `kABPersonCompositeNameFormatFirstNameFirst`

See https://developer.apple.com/reference/addressbook/1619730-sort_order

File: Signal/src/views/ContactTableViewCell.m [hunk 2]
```diff
@@ -30,34 +30,53 @@ - (NSAttributedString *)attributedStringForContact:(Contact *)contact {
     NSMutableAttributedString *fullNameAttributedString =
         [[NSMutableAttributedString alloc] initWithString:contact.fullName];
 
+    BOOL firstNameDisplay = ABPersonGetCompositeNameFormat() == kABPersonCompositeNameFormatFirstNameFirst ? YES : NO;
```
Comment (michaelkirk) [inline Signal/src/views/ContactTableViewCell.m]: `BOOL displayGivenNameFirst = ABPersonGetCompositeNameFormat() == kABPersonCompositeNameFormatFirstNameFirst`

File: Signal/src/views/ContactTableViewCell.m (hunk 1 above)
Comment (michaelkirk) [inline Signal/src/views/ContactTableViewCell.m]: `BOOL sortByFirstName` would be clearer

**Comment by michaelkirk** [comment]
It seems to me like it would be clearer to build two [attributed strings passing the attributes in a dictionary](https://developer.apple.com/library/ios/documentation/Cocoa/Reference/Foundation/Classes/NSAttributedString_Class/#//apple_ref/occ/instm/NSAttributedString/initWithString:attributes:), then concatenating the strings, rather than dealing with the NSRanges/length calculations.

File: Signal/src/views/ContactTableViewCell.m (hunk 2 above)
Comment (michaelkirk) [inline Signal/src/views/ContactTableViewCell.m]: Should be `kABPersonSortByFirstName`

**Comment by abcdev** [comment]
@michaelkirk I tried to stay as close as possible to the original code. But I could change that of course.

**Pull request closed without merging**